cel/program.go

```diff
@@ -67,7 +67,7 @@ func (ed *evalDetails) State() interpreter.EvalState {
 
 // prog is the internal implementation of the Program interface.
 type prog struct {
```

Comment:
Do you want to de-interface-ify Program too?

Also how about Source and EvalDetails too?

Reply:
I end up treating the `Source` as an interface elsewhere and it works out pretty well since it's used to provide location information for the input.

I will de-interface `EvalDetails`. I think I had multiple implementations for `Program`, but if not I'll de-interface it too.